animating-widgets.md

```diff
@@ -0,0 +1,293 @@
+to the animation’s values by using the
+[`addListener()`](https://docs.flutter.io/flutter/animation/Animation/addListener.html)
+method. The framework invokes the listener callback whenever the value of the
+animation changes. Typically, you’ll want to call the
+[`setState()`](http://docs.flutter.io/flutter/material/State/setState.html)
+method when the animation value change occurs, to trigger the framework to
+rebuild the widget tree.
+
+The following code snippet shows how you might create a simple animation
+effect to make an logo image appear to expand.
+
+```dart
+import 'package:flutter/material.dart';
+import 'package:flutter/animation.dart';
+
+class LogoApp extends StatefulWidget {
+  LogoAppState createState() => new LogoAppState();
+}
+
+class LogoAppState extends State<LogoApp> {
+  Animation<double> animation;
+
+  initState() {
+    AnimationController controller = new AnimationController(
+      duration: const Duration(milliseconds: 5000)
+    );
+    animation = new Tween(begin: 0.0, end: 200.0).animate(controller);
+    animation.addListener(() {
+      setState(() {
+      // the state that has changed here is the animation object’s value
+      });
+    });
+    controller.forward();
+    super.initState();
+  }
+
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Container(
+        margin: new EdgeInsets.symmetric(vertical: 10.0),
+        height: animation.value,
+        width: animation.value,
+        child: new AssetImage(
+          name: 'assets/flutter_logo.png',
+          fit: ImageFit.contain
+        )
+      )
+    );
+  }
+}
+
+void main() {
+  runApp(
+    new MaterialApp(
+      title: 'My Animation Test',
+      routes: <String, WidgetBuilder> {
+        '/': (BuildContext context) => new LogoApp()
+      }
+    )
+  );
+}
+```
+
+You can simplify the code snippet above further by using the
+[`AnimatedWidget`](http://docs.flutter.io/flutter/material/AnimatedWidget-class.html)
+class. This class hides the state object and listener you previously had to
+implement to run the animation.
+
+```dart
+import 'package:flutter/material.dart';
+import 'package:flutter/animation.dart';
+
+class LogoApp extends AnimatedWidget {
+  LogoApp({Key key, Animation<double> animation})
+      : super(key: key, animation: animation);
+
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Container(
+        margin: new EdgeInsets.symmetric(vertical: 10.0),
+        height: animation.value,
+        width: animation.value,
+        child: new AssetImage(
+           name: 'assets/flutter_logo.png',
+           fit: ImageFit.contain
+        )
+      )
+    );
+  }
+}
+
+void main() {
+  final AnimationController controller = new AnimationController(
+    duration: const Duration(milliseconds: 5000)
+  );
+  final animation = new Tween(
+    begin: 0.0,
+    end: 200.0)
+  .animate(controller);
```

Comment:
I'm not sure this is a good example.  It's extremely unusual to create an AnimationController inside the `main` function.  Also, now with the `vsync` parameter, you'll have trouble figuring out which `vsync` you want to attach your animation to.

I'm not sure it's worth presenting these two examples.  Instead, we should just use `AnimatedBuilder` in the first example.  That's the code you'd normally write and it's simple enough to explain in one go.